crates/bitwarden/src/mobile/vault/client_ciphers.rs

```diff
@@ -24,6 +24,11 @@ impl<'a> ClientCiphers<'a> {
             cipher_view.generate_cipher_key(key)?;
         }
 
+        // For compatibility reasons, we only create checksums for ciphers that have a key
+        if cipher_view.key.is_some() {
+            cipher_view.generate_checksums();
+        }
```

Comment:
I think this should move into the cipher logic. This is not temporary behaviour, and I believe we could easily do this as part of the encyrpt operation?

Reply:
Yeah, my thought was to group it with the cipher key behaviour and eventually when the flag was removed move both into the encrypt operation, but we can do it now with the checksums. I've updated the PR